PSModule/AlmLabMgmtClient/SDK/Handler/LabPollHandler.cs

```diff
@@ -0,0 +1,144 @@
+        private const string START_TIME = "start-time";
+        private const string STATE = "state";
+        private const string COMPLETED_SUCCESSFULLY = "completed-successfully";
+        private const string RESERVATION_ID = "reservation-id";
+
+        private EventLogHandler _eventLogHandler;
+
+        public LabPollHandler(IClient client, string entityId) : base(client, entityId)
+        {
+        }
+
+        public LabPollHandler(IClient client, string entityId, int interval) : base(client, entityId, interval) { }
+
+        protected override async Task<bool> DoPoll()
+        {
+            bool ok = false;
+
+            Response res = await GetRunEntityData();
+            if (res.IsOK)
+            {
+                SetTimeslotId(res);
+                _eventLogHandler = new EventLogHandler(_client, _timeslotId);
+                if (!_timeslotId.IsNullOrWhiteSpace())
+                {
+                    ok = await base.DoPoll();
+                }
+            }
+            else
+            {
+                LogPollingError(res);
+            }
+            return ok;
+        }
+
+        protected override async Task<Response> GetResponse(bool logRequestUrl)
+        {
+            return await new PollAlmLabMgmtRunRequest(_client, _runId).Execute(logRequestUrl);
+        }
+
+        protected async override Task LogProgress(bool logRequestUrl)
+        {
+            await _eventLogHandler.Log(logRequestUrl);
+        }
+
+        protected override bool IsFinished(Response response)
+        {
+            bool ret = false;
+            try
+            {
+                string xml = response.ToString();
+                string endTime = Xml.GetAttributeValue(xml, END_TIME);
+                if (!endTime.IsNullOrWhiteSpace())
+                {
+                    string startTime = Xml.GetAttributeValue(xml, START_TIME);
+                    string currentRunState = Xml.GetAttributeValue(xml, STATE);
+                    _logger.LogInfo($"Timeslot {_timeslotId} is [{currentRunState}].\nRun start time: [{startTime}], Run end time: [{endTime}]");
+                    ret = true;
+                }
+            }
+            catch (ThreadInterruptedException)
+            {
+                throw;
+            }
+            catch
+            {
+                _logger.LogError($"Failed to parse response: {response}");
+                ret = true;
+            }
+
+            return ret;
+        }
+
+        protected override bool LogRunEntityResults(Response response)
+        {
+            bool ok = false;
+            try
+            {
+                string xml = response.ToString();
+                string state = Xml.GetAttributeValue(xml, STATE);
+                string completedSuccessfully = Xml.GetAttributeValue(xml, COMPLETED_SUCCESSFULLY);
+                _logger.LogInfo($"Run state of {_runId}: {state}, Completed successfully: {completedSuccessfully}");
+                ok = true;

```

Comment:
Before setting `ok` to `true`, do we need check if `completedSuccessfully` is one of the values that we treat it as success?

Reply:
The function LogRunEntityResults should return true if the response is valid (contains the expected attributes, regardless their value). The real test result will be queried and printed later using LabPublisher.cs.

Reply:
You mentioned **valid**, however, the variable `state` and `completedSuccessfully` are never checked. For example, if `completedSuccessfully` is empty?